File: Citologia/web/app/model/TarjetaPrueba.js
Ext.define('Citologia.model.TarjetaPrueba', {
    extend: 'Ext.data.Model',
    fields: ['idTarjeta', 'paciente', 'antecedente', 'primeraCitologia', 'tipoCaso', 'nombre', {name: 'enfermedadesTransmisionSexual', defaultValue: new Array()}],
    idProperty: 'idTarjeta',
    associations: [{
            type: 'hasMany',
            model: 'Citologia.model.EnfermedadTransmisionSexual', name: 'enfermedadesTransmisionSexuales', associationKey: 'enfermedadesTransmisionSexual'
        }, {
            type: 'belongsTo',
            model: 'Citologia.model.DiagnosticoFinal', getterName: 'getDiagnostico', setterName: 'setDiagnostico', associationKey: 'primeraCitologia.diagnosticoFinal'
        }, {
            type: 'hasMany',
            model: 'Citologia.model.ResponsableMuestra', name: 'responsables', associationKey: 'primeraCitologia.responsablesMuestras'
        }, {
            type: 'belongsTo',
            model: 'Citologia.model.TipoCaso', associationKey: 'tipoCaso', getterName: 'getTipoCaso', setterName: 'setTipoCaso'
        }, {
            type: 'belongsTo',
            model: 'Citologia.model.TipoAnticonceptivo', associationKey: 'antecedente.tipoAnticonceptivo', getterName: 'getTipoAnticonceptivo', setterName: 'setTipoAnticonceptivo'
        }]
});

File: Citologia/web/WEB-INF/Citologia/src/entidades/TarjetaPrueba.java
package entidades;

public class TarjetaPrueba {
	private Integer idTarjeta;

	private Paciente paciente;

	private Antecedente antecedente;

	private PrimeraCitologia primeraCitologia;

	private TipoCaso tipoCaso;

	private DiagnosticoFinal diagnosticoFinal;

	private String nombre;

	public Integer getIdTarjeta() {
		return idTarjeta;
	}

	public void setIdTarjeta(Integer idTarjeta) {
		this.idTarjeta = idTarjeta;
	}

	public Paciente getPaciente() {
		return paciente;
	}

	public void setPaciente(Paciente paciente) {
		this.paciente = paciente;
	}

	public Antecedente getAntecedente() {
		return antecedente;
	}

	public void setAntecedente(Antecedente antecedente) {
		this.antecedente = antecedente;
	}

	public PrimeraCitologia getPrimeraCitologia() {
		return primeraCitologia;
	}

	public void setPrimeraCitologia(PrimeraCitologia primeraCitologia) {
		this.primeraCitologia = primeraCitologia;
	}

	public TipoCaso getTipoCaso() {
		return tipoCaso;
	}

	public void setTipoCaso(TipoCaso tipoCaso) {
		this.tipoCaso = tipoCaso;
	}

	public DiagnosticoFinal getDiagnosticoFinal() {
		return diagnosticoFinal;
	}

	public void setDiagnosticoFinal(DiagnosticoFinal diagnosticoFinal) {
		this.diagnosticoFinal = diagnosticoFinal;
	}

	public String getNombre() {
		nombre = paciente.getNombreCompleto();
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = paciente.getNombreCompleto();
	}

}
